src/main/scala/edu/ie3/simona/event/listener/ResultEventListener.scala

```diff
@@ -117,14 +114,19 @@ object ResultEventListener extends Transformer3wResultSupport {
               )
               .flatMap { fileName =>
                 if (fileName.endsWith(".csv") || fileName.endsWith(".csv.gz")) {
-                  ResultEntityCsvSink(
-                    fileName.replace(".gz", ""),
-                    new ResultEntityProcessor(resultClass),
-                    fileName.endsWith(".gz")
-                  ).map((resultClass, _))
+                  Future {
```

Comment:
Does this work as expected when we have multiple `ResultEventListener`? We hand back an Iterable of Futures which we then map to `SinkResponse`. And as it seems to me we don't wait for every sink to be finished but are going to idle unstashing all messages as soon as we get the first `SinkResponse`

Reply:
No, the iterable of Futures is still handed the same way. Difference is: `ResultEntityCsvSink.apply` returns the sink directly, not a future to one. Thus, we need to wrap in a future here.